File: Tema3/Ejercicio1/app/src/main/java/com/example/ejercicio2/MainActivity.java
package com.example.ejercicio2;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;
import androidx.appcompat.widget.Toolbar;

public class MainActivity extends AppCompatActivity {

    private Toolbar toolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        toolbar = findViewById(R.id.toolbar);

        toolbar.inflateMenu(R.menu.main);
        toolbar.setOnMenuItemClickListener(item -> {
            if(item.getItemId() == R.id.action_search) launchFilter();
            return onContextItemSelected(item);
        });

    }

    public void launchPlay(View view){
        buttonPressed();
        Intent i = new Intent(this, Play.class);
        startActivity(i);
    }

    public void launchNewPlayer(View view){
        buttonPressed();
        Intent i = new Intent(this, NewPlayer.class);
        startActivity(i);
    }

    public void launchPreferences(View view){
        buttonPressed();
        Intent i = new Intent(this, Preferences.class);
        startActivity(i);
    }

    public void launchFilter(){
        Intent i = new Intent(this, Filter.class);
        startActivity(i);
    }
    
    private void buttonPressed(){
        Toast.makeText(this, "Has pulsado el material button", Toast.LENGTH_SHORT).show();
    }
}

File: Practica5_2/app/src/main/java/com/example/practica5_2/MainActivity.kt
package com.example.practica5_2

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.navigation.NavController
import androidx.recyclerview.widget.GridLayoutManager
import com.example.practica5_2.adapters.CardViewAdapter
import com.example.practica5_2.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private val imageList = arrayListOf(
        R.drawable.corona_solar,
        R.drawable.manchasolar,
        R.drawable.magnetosfera,
        R.drawable.filamentos,
        R.drawable.espiculas,
        R.drawable.erupcionsolar
    )
    private val titleList = arrayListOf("corona solar", "mancha solar", "magnetosfera", "filamentos", "espiculas", "erupcionsolar")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        binding.recycler.layoutManager = GridLayoutManager(this@MainActivity, 2)
        binding.recycler.adapter = CardViewAdapter(imageList, titleList)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.main_menu, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val intent = Intent(this@MainActivity, ComparaPlanetas::class.java)
        startActivity(intent)
        return super.onOptionsItemSelected(item)
    }
}